ergotree-interpreter/src/eval/decode_point.rs

```diff
@@ -0,0 +1,52 @@
+use ergotree_ir::mir::decode_point::DecodePoint;
+//use ergotree_ir::mir::constant::TryExtractInto;
+use ergotree_ir::mir::value::Value;
+
+use crate::eval::env::Env;
+use crate::eval::EvalContext;
+use crate::eval::EvalError;
+use crate::eval::EvalError::Misc;
+use crate::eval::Evaluable;
+use ergotree_ir::mir::constant::TryExtractInto;
+use ergotree_ir::sigma_protocol::dlog_group;
+
+impl Evaluable for DecodePoint {
+    fn eval(&self, env: &Env, ctx: &mut EvalContext) -> Result<Value, EvalError> {
+        let point_bytes = self.input.eval(env, ctx)?.try_extract_into::<Vec<u8>>()?;
+        let point = dlog_group::from_bytes(point_bytes)
+            .map_err(|_| Misc(String::from("Invalid EC point")))?;
+        Ok(point.into())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::rc::Rc;
+
+    use crate::eval::context::Context;
+    use crate::eval::tests::eval_out;
+
+    use super::*;
+
+    use ergotree_ir::mir::expr::Expr;
+    use proptest::collection;
+    use proptest::prelude::*;
+    use sigma_test_util::force_any_val;
+
+    proptest! {
+
+        #[test]
+        fn eval(bytes in collection::vec(any::<u8>(), 33)) {
```

Comment:
`EcPoint` has `Arbitrary` implemented so the test could be simplified to something like:
```
fn eval(ecp in any::<EcPoint>()) {
    let bytes = ecp.sigma_serialize_bytes();
    let expr: Expr = DecodePoint {input: Expr::Const(bytes.clone().into()).into()}.into();
    let res = eval_out_wo_ctx::<dlog_group::EcPoint>(&expr);
    prop_assert_eq!(res, ecp);
}